### PR TITLE
facility sort,search household,and app relaunch data load removed

### PR DESCRIPTION
**Pull request by vishal-egov**

### CODE REVIEW

File: apps/health_campaign_field_worker_app/lib/blocs/facility/facility.dart [hunk 1]
```diff
@@ -40,6 +40,12 @@ class FacilityBloc extends Bloc<FacilityEvent, FacilityState> {
           version: 1,
           fields: [const AdditionalField('type', 'ProvisionalWarehouse')],
         ),
+        auditDetails: AuditDetails(
+          createdBy: 'createdBy',
```
Comment (naveen-egov) [inline apps/health_campaign_field_worker_app/lib/blocs/facility/facility.dart:44]: @vishal-egov  Pls add the createdBy and updatedBy

File: apps/health_campaign_field_worker_app/lib/blocs/facility/facility.dart [hunk 2]
```diff
@@ -71,6 +77,11 @@ class FacilityBloc extends Bloc<FacilityEvent, FacilityState> {
       final facility = facilities
           .firstWhereOrNull((element) => element.id == latestStock?.facilityId);
 
+      allFacilities.sort((a, b) => b.auditDetails!.lastModifiedTime
+          .compareTo(a.auditDetails!.lastModifiedTime));
+      facilities.sort((a, b) => b.auditDetails!.lastModifiedTime
```
Comment (naveen-egov) [inline apps/health_campaign_field_worker_app/lib/blocs/facility/facility.dart:82]: @vishal-egov  add a null check

File: apps/health_campaign_field_worker_app/lib/data/repositories/local/facility.dart [hunk 3]
```diff
@@ -76,6 +76,12 @@ class FacilityLocalRepository extends FacilityLocalBaseRepository {
                 type: address.type,
                 rowVersion: address.rowVersion,
               ),
+        auditDetails: AuditDetails(
+          createdBy: facility.auditCreatedBy!,
```
Comment (naveen-egov) [inline apps/health_campaign_field_worker_app/lib/data/repositories/local/facility.dart:80]: add a null check of auditDetails

**Review by naveen-egov — CHANGES_REQUESTED** [review]
pls make the changes

File: apps/health_campaign_field_worker_app/lib/blocs/facility/facility.dart (hunk 1 above)
Comment (vishal-egov) [reply apps/health_campaign_field_worker_app/lib/blocs/facility/facility.dart:44]: @naveen-egov this is a dummy warehouse we areadding so there is no createdby or lastmodified for this facility, had to add it becase it i mandatory

File: apps/health_campaign_field_worker_app/lib/data/repositories/local/facility.dart (hunk 3 above)
Comment (vishal-egov) [reply apps/health_campaign_field_worker_app/lib/data/repositories/local/facility.dart]: Done

File: apps/health_campaign_field_worker_app/lib/blocs/facility/facility.dart (hunk 2 above)
Comment (vishal-egov) [reply apps/health_campaign_field_worker_app/lib/blocs/facility/facility.dart]: Moved to a different place and added null check there

File: apps/health_campaign_field_worker_app/lib/data/repositories/local/facility.dart [hunk 4]
```diff
@@ -76,6 +76,12 @@ class FacilityLocalRepository extends FacilityLocalBaseRepository {
                 type: address.type,
                 rowVersion: address.rowVersion,
               ),
+        auditDetails: AuditDetails(
+          createdBy: facility.auditCreatedBy ?? "",
```
Comment (naveen-egov) [inline apps/health_campaign_field_worker_app/lib/data/repositories/local/facility.dart:80]: retun null value is it's not present

File: apps/health_campaign_field_worker_app/lib/blocs/facility/facility.dart (hunk 1 above)
Comment (naveen-egov) [reply apps/health_campaign_field_worker_app/lib/blocs/facility/facility.dart:44]: add the user id  instead of empty string

**Review by naveen-egov — CHANGES_REQUESTED** [review]

**Pull request merged**